### ISSUE TITLE
[ENH] Add various Kolmogorov-Arnold Network using PyKAN

### ISSUE THREAD

**Issue by benHeid** [sktime/sktime#6390]
**Is your feature request related to a problem? Please describe.**
This issue proposes to add support for pyKAN based models

Potential Estimators:

* [ ] PyKAN based forecaster (simple forward KAN)
* [ ] PyKAN recurrent forecaster (should be simple to implement)
* [ ] PyKAN classifier (should probably be a recurrent network?)

**Comment by fkiraly** [comment]
we will also need to discuss how to deal with the high runtimes, e.g., in tests

**Comment by astrogilda** [comment]
@fkiraly is this potentially solvable by using `uv`?

**Comment by fkiraly** [comment]
no, I think the runtime comes from the `pykan` estimators directly, not the environment setup.

**Comment by fkiraly** [comment]
that might work, if the parameters allow that kind of control
